types/vote_set.go

```diff
@@ -194,6 +194,7 @@ func (voteSet *VoteSet) addVote(vote *Vote) (added bool, err error) {
 	}
 
 	// Check signature.
+	// TODO(thane): Should we always require vote extensions in vote sets?
```

Comment:
I think we should, as long as they are precommits
The only case we don't want precommits to have extensions is when embedded into an evidence, which is what complicates this problem.
Would it make sense to also requires extensions in evidences (as long as they refer to precommits) ?
If yes, then all this logic would be greatly simplified

Reply:
It doesn't look like `VoteSet` is used in relation to evidence. I've now updated the code to use `ValidateWithExtension` instead and it seems to work, but we'll see when we enforce the vote extension requirement in a subsequent PR.